### PR TITLE
Feature/ci

### PR DESCRIPTION
**Pull request by guyeisenbach**

### CODE REVIEW

File: .github/workflows/ci_e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    name: Extract supported_features
+    outputs:
+      supported-features: ${{ steps.supported-features.outputs.value }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:19]: 20.x

File: .github/workflows/ci_e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  extract_metadata:
+    runs-on: ubuntu-latest
+    name: Extract supported_features
+    outputs:
+      supported-features: ${{ steps.supported-features.outputs.value }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:17]: v4

File: .github/workflows/ci_e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - extract_metadata
+
+    steps:
+
+      - name: Checkout Repo
+        uses: actions/checkout@v3
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:40]: v4

File: .github/workflows/ci_e2e.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+      - name: Build local cluster
+        run: ./ci_files/build_cluster.sh
+
+      - name: Build Enforcer Docker image
+        run: |
+          docker build . -t localhost:5001/java-enforcer-sample-site:$SAMPLE_SITE_IMAGE_TAG && \
+          docker push localhost:5001/java-enforcer-sample-site:$SAMPLE_SITE_IMAGE_TAG
+
+      - uses: azure/setup-helm@v3
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:53]: v4

File: .github/workflows/ci_e2e.yaml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        with:
+          version: '3.14.1'
+
+      - name: Clone helm charts repo - mock-collector
+        uses: actions/checkout@v3
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:58]: v4

File: .github/workflows/ci_e2e.yaml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        with:
+          repository: PerimeterX/connect-helm-charts
+          token: ${{ secrets.CONNECT_PULL_TOKEN }}
+          ref: enforcer-spec-tests-0.7.1
+          path: ./deploy_charts/enforcer-spec-tests
+
+
+      - name: Clone helm charts repo - sample-site
+        uses: actions/checkout@v3
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:76]: v4

File: .github/workflows/ci_e2e.yaml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        with:
+          repository: PerimeterX/connect-helm-charts
+          token: ${{ secrets.CONNECT_PULL_TOKEN }}
+          ref: sample-site-0.4.0
+          path: ./deploy_charts/sample-site
+
+      - name: Set up Google Cloud SDK
+        id: 'auth'
+        uses: 'google-github-actions/auth@v1'
```
Comment (chen-zimmer-px) [inline .github/workflows/ci_e2e.yaml:85]: v2

**Review by chen-zimmer-px — CHANGES_REQUESTED** [review]
Please change the versions

File: .github/workflows/ci_e2e.yaml (hunk 1 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 2 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+jobs:
+
+  extract_metadata:
+    runs-on: ubuntu-latest
+    name: Extract supported_features
+    outputs:
+      supported-features: ${{ steps.supported-features.outputs.value }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 3 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 4 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 5 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 6 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

File: .github/workflows/ci_e2e.yaml (hunk 7 above)
Comment (guyeisenbach) [reply .github/workflows/ci_e2e.yaml]: done

**Review by chen-zimmer-px — APPROVED** [review]

**Pull request merged**